src/main/java/com/uber/cadence/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -576,70 +599,68 @@ public void childWorkflowStarted(ChildWorkflowExecutionStartedEventAttributes a)
   }
 
   @Override
-  public void childWorklfowFailed(String activityId, ChildWorkflowExecutionFailedEventAttributes a)
+  public void childWorklfowFailed(
+      String activityId, ChildWorkflowExecutionFailedEventAttributes a, long initiatedEventId)
       throws InternalServiceError, EntityNotExistsError, BadRequestError {
     update(
         ctx -> {
           String childId = a.getWorkflowExecution().getWorkflowId();
-          StateMachine<ChildWorkflowData> child = getChildWorkflow(childId);
+          StateMachine<ChildWorkflowData> child = getChildWorkflow(initiatedEventId);
           child.action(StateMachines.Action.FAIL, ctx, a, 0);
-          childWorkflows.remove(childId);
+          childWorkflows.remove(a.getInitiatedEventId());
```

Comment:
Seems like attribute has eventID in it. Can we just always use that and avoid the interface change?

Reply:
Good point. It looks like it was leftover from the original version of the refactoring.